File: ProductsSolution/Entities/Sale.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Entities
{
    public class Sale : BaseEntity
    {
        public int ProductId { get; set;}

        [ForeignKey("ProductId")]
        public Product Product { get; set; }

        public int Amount { get; set;}
        public DateTime Date { get; set;}
        public string str_date { get; set; }

        public int UserId { get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; }

        public int SalePointId { get; set; }
        [ForeignKey("SalePointId")]
        public SalePoint SalePoint { get; set; }

        public int month { get; set; }
        public int year { get; set; }
        public int day { get; set; }
    }
}

File: angular-client/src/app/Model/Sale.ts

export class Sale {
  public id: number;
  public productId: string;
  public salePointId: number;
  public amount: number;
  public date: Date;
  public productDescription: string;
  public SalePointDescription: string;
  public month: string;
  public year: string;

  // tslint:disable-next-line:max-line-length
  constructor(id: number, productId: string, salePointId: number,  amount: number, date: Date, productDescription: string,
              salePointDescription: string, month: string, year: string ) {
    this.id = id;
    this.productId = productId;
    this.salePointId = salePointId;
    this.amount = amount;
    this.date = date;
    this.productDescription = productDescription;
    this.SalePointDescription = salePointDescription;
    this.month = month;
    this.year = year;
  }
}
